### ISSUE TITLE
System low battery fault not reported as state

### ISSUE THREAD

**Issue by ptr727** [ufodone/envisalink_new#61]
We had a whole day power outage, drained the battery completely, been a week and battery is still low, new battery ordered.  
Note, this may be related to the #17 discussion on not using attributes as most HA integrations report sensors or discrete values.  

Problem is that I cannot find a state/attribute reported in HA that reports the low battery fault condition.  
I can see that the keypad state occasionally changes to report low battery, and on the EyezOn site the fault is reported as low battery, but not in an active attribute/sensor to report on.

Am I missing something, or is this an enhancement?

<img width="764" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/1a5e92ad-7259-4c26-acc4-0263b2aad5cf">
<img width="249" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/859bb269-da52-4121-b899-5e631b038f7d">

Note in HA pic `Bat trouble false`, and Logbook `Home Alarm Keypad changed to Zone/Sensor Low Battery`.

**Comment by ufodone** [comment]
I'm not the original author of the integration but it looks like it should respond to low battery updates from the EVL and update the "Bat trouble" attribute on the keypad accordingly.  I was able to confirm this with an EVL simulator I built but don't have an easy way to test it against a real system.

If you alarm is still in this state, could you enable debug logging on the integration, restart HA and send me the logs after it's been running for a bit.  That should show what you EVL is sending to the integration and may provide some insight into the issue.

You can turn logging on by adding something like below to your configuration.yaml:

```
logger:
  default: info
  logs:
    custom_components.envisalink_new: debug
```

**Comment by ptr727** [comment]
<img width="761" alt="image" src="https://github.com/ufodone/envisalink_new/assets/2061579/f341c480-9510-4d9c-bf4b-c5beff87bae3">

[home-assistant.envisalink.log](https://github.com/ufodone/envisalink_new/files/13227815/home-assistant.envisalink.log)

**Comment by ufodone** [comment]
The Envisalink TPI documentation indicates that an 800 notification (Panel Battery Trouble) should be sent by the device when the panel's battery is low.  I don't see that in your log file.  I do see an 849 update which is what is triggering the `Zone/Sensor Low Battery` keypad message - but according to the documentation, that is meant to indicate a low battery issue on a Zone rather than the panel itself.

This integration is wired up to handle the 800 messages and update the `bat_trouble` attribute accordingly but there's likely nothing that can be done if the Envisalink itself isn't sending the updates.

Are you running the latest firmware from EyezOn?

**Comment by ptr727** [comment]
Firmware: `Firmware Version: 01.12.205` which I [understand](https://forum.eyezon.com/viewtopic.php?t=5115) to be latest for EVL3.

**Comment by rct** [comment]
In case this helps,  a few weeks ago during some testing I disconnected the battery from the panel. I did get an 800 and 849 (EVL4, panel is an 1832).  The `bat trouble` attribute did change. I use a template to turn that into it's own entity and send an alert based on that. 

Unfortunately, I realized I'm currently running the old integration (the code bundled with hass.  I thought I was running the newer code via a custom integration, but I no longer seem to have that configured for some reason.

This is the EVL TPI traffic I captured via the alarmserver proxy
```
Nov 02 15:34:24 - DEBUG - core/envisalink.py:handle_line@158: RX < 650 - Partition Home Ready
Nov 02 15:34:24 - DEBUG - core/state.py:update@46: Discarded event. State not changed. (partition 1)
Nov 02 15:34:43 - DEBUG - core/envisalink.py:envisalink_proxy@278: PROXY > 00090
Nov 02 15:34:43 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "50000025"
Nov 02 15:34:43 - DEBUG - core/envisalink.py:handle_line@158: RX < 500 - Command Acknowledge 000
Nov 02 15:35:16 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "5109100"
Nov 02 15:35:16 - DEBUG - core/envisalink.py:handle_line@158: RX < 510 - Keypad Led State - Partition 1
Nov 02 15:35:16 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "8401CD"
Nov 02 15:35:16 - DEBUG - core/envisalink.py:handle_line@158: RX < 840 - Partition Home Trouble LED ON
Nov 02 15:35:20 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "51110F8"
Nov 02 15:35:20 - DEBUG - core/envisalink.py:handle_line@158: RX < 511 - Keypad Led Flash State - Partition 1
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "8490106"
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@158: RX < 849 - Verbose Trouble Status 01
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "51080FE"
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@158: RX < 510 - Keypad Led State - Partition 1
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "51100F7"
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@158: RX < 511 - Keypad Led Flash State - Partition 1
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "5109100"
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@158: RX < 510 - Keypad Led State - Partition 1
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "6501CC"
Nov 02 15:35:21 - DEBUG - core/envisalink.py:handle_line@158: RX < 650 - Partition Home Ready
Nov 02 15:35:43 - DEBUG - core/envisalink.py:envisalink_proxy@278: PROXY > 00090
Nov 02 15:35:43 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "50000025"
Nov 02 15:35:43 - DEBUG - core/envisalink.py:handle_line@158: RX < 500 - Command Acknowledge 000
Nov 02 15:36:35 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "6501CC"
Nov 02 15:36:35 - DEBUG - core/envisalink.py:handle_line@158: RX < 650 - Partition Home Ready
Nov 02 15:36:43 - DEBUG - core/envisalink.py:envisalink_proxy@278: PROXY > 00090
Nov 02 15:36:43 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "50000025"
Nov 02 15:36:44 - DEBUG - core/envisalink.py:handle_line@158: RX < 500 - Command Acknowledge 000
Nov 02 15:37:04 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "51081FF"
Nov 02 15:37:04 - DEBUG - core/envisalink.py:handle_line@158: RX < 510 - Keypad Led State - Partition 1
Nov 02 15:37:05 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "8411CE"
Nov 02 15:37:05 - DEBUG - core/envisalink.py:handle_line@158: RX < 841 - Partition Home Trouble LED OFF
Nov 02 15:37:43 - DEBUG - core/envisalink.py:envisalink_proxy@278: PROXY > 00090
Nov 02 15:37:44 - DEBUG - core/envisalink.py:handle_line@131: RX RAW < "50000025"
```

**Comment by ufodone** [comment]
I tried disconnecting the battery on my own panel and I also saw the `bat_trouble` attribute change.

@ptr727 Given that your logs didn't show the updates which trigger the `bat_trouble` change and that this appears to be working for @rct and me on EVL 4 units, I wonder if this is a limitation of the EVL3 or potentially the specific alarm panel.  

Unfortunately, if the EVL doesn't send the requisite updates, there's nothing I can do in the HA integration to fix this.
